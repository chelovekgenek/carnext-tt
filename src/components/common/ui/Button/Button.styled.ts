import styled from "styled-components"

export const Button = styled.button`
  background: linear-gradient(45deg, rgba(237, 135, 0, 1) 0%, rgba(238, 76, 20, 1) 100%);
  font-size: medium;
  font-weight: bold;
  border: none;
  border-radius: 2rem;
  color: white;
  padding: 1rem 2rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  text-transform: uppercase;
  cursor: pointer;
  &:hover {
    background: linear-gradient(45deg, rgba(238, 76, 20, 1) 0%, rgba(237, 135, 0, 1) 100%);
  }
  &:disabled {
    background: lightgray;
  }
  i.fas {
    margin-left: 1rem;
  }
`
