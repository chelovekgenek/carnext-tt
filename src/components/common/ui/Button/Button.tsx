import React, { ButtonHTMLAttributes } from "react"

import * as Styles from "./Button.styled"

interface IProps extends ButtonHTMLAttributes<never> {}

export const Button: React.FC<IProps> = props => <Styles.Button {...props} />
