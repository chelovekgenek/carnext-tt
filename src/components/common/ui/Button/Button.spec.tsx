import React from "react"
import { shallow } from "enzyme"

import { Button } from "./Button"
import * as Styled from "./Button.styled"

describe("Layout", () => {
  const render = (props: React.ComponentProps<typeof Button>) => shallow(<Button {...props} />)
  it("should render", () => {
    expect(render({})).toMatchSnapshot()
  })
  it("should render with passed button props", () => {
    const props = { onClick: jest.fn(), type: "submit" as const }
    const element = render(props).find(Styled.Button)
    element.simulate("click")
    expect(props.onClick).toBeCalled()
    expect(element.prop("type")).toBe("submit")
  })
  it("should render with children", () => {
    expect(render({ children: "text" }).find(Styled.Button)).toMatchSnapshot()
  })
})
