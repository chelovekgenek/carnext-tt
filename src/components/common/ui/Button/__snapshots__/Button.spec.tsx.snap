// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render 1`] = `<styled.button />`;

exports[`Layout should render with children 1`] = `
<styled.button>
  text
</styled.button>
`;
